main.go

```diff
@@ -812,6 +812,35 @@ func (task *TaskRun) ExecuteCommand(index int) *CommandExecutionError {
 	if err != nil {
 		return WorkerShutdown(err)
 	}
+
+	// the duration until the expiry is precisely -1 *
+	// the amount of time since the expiry
+	duration := -time.Since(time.Time(task.Definition.Expires))
+	commandLogURL, err := Queue.GetArtifact_SignedURL(
```

Comment:
Wouldn't this cause an expiration issue later? You're using a signed url in the redirect artifact.